filebeat/input/log/input.go

```diff
@@ -284,6 +294,10 @@ func (p *Input) Run() {
 			p.cleanupStates()
 		}
 	}
+
+	if p.healthy {
+		p.updateStatus(status.Running, "finished the scan")
+	}
```

Comment:
Why do we need to report this? I think it's going to be too much traffic if we report each scan. Additionally, if we report something finished there has to be a report that something started before that.

Reply:
> I think it's going to be too much traffic if we report each scan.

That's true. But, the state will be propagated to agent only if the current state different than previous state. 
If we face any transient errors, then we need to change the state back to healthy once the error is resolved. I think placing the check at the end would suffice.

> Additionally, if we report something finished there has to be a report that something started before that.

It's done here https://github.com/elastic/beats/blob/d40a9e37285850a28ee95c474d4c9b3ac9e9e792/x-pack/libbeat/management/managerV2.go#L407-L408
for all agent units once they're added.